packages/tools/devtools/devtools-core/src/data-visualization/DefaultVisualizers.ts

```diff
@@ -25,6 +25,16 @@ import {
 	VisualTreeNode,
 } from "./VisualTree";
 
+/**
+ * @public
+ */
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+export class SharedTreeObject {
```

Comment:
I don't think we want to export this here. If this factory object isn't available in SharedTree's API, we should probably add it. This is a core pattern available in other DDS packages. @noencke is there any reason we can't expose the factory statics in SharedTree?

Reply:
Agreed it probably shouldn't be exported. 

The `container.create([DDS_Class])` function is oddly specific in that it requires not just a static factory method on the given class object but also the class object must have a constructor which returns a type with a `handle` field. As far as we have seen, that constructor is never actually invoked. The API typing seems to be attempting to force the user to provide the DDS class object rather than, say, just asking for a function which produces instances of that class. I'm not sure why it is like this; perhaps it is trying to keep people from making mistakes using the API? The actual class `SharedTree` is intentionally not exposed; we provide an interface `ISharedTree` that reduces the API surface to what we want (otherwise, all kinds of types get exposed which should stay internal). 

We anticipate that going forward, the API for consuming these factories will need to change eventually for an additional reason: they don't allow for passing arguments to the DDS creation. SharedTree is going to need that, and really any DDS could/should (e.g. passing your DDS a "summary version" after you've done a data format migration). Without having that, even if we exposed the SharedTree class, apps would have to make subclasses of SharedTree anyway which overwrite the kind of _factory_ that they expose, where that factory is hardcoded to provide the config that is passed into the trees at construction time. This is what some customers of the legacy SharedTree had to do.

I think this is a conversation that needs to continue so as to inform the evolution of the initialization APIs. In the meantime, since both "make this little wrapper class and do a cast" and "expose the SharedTree class" are temporary workarounds, we opted for the former since it's much, much easier.

Reply:
cc @Abe27342 since I know you've mentioned being able to pass arguments to DDSes.

Reply:
Moved the class to `experimental/dds/tree2/src/shared-tree/sharedTree.ts`.

Reply:
I would love to see a larger conversation around this. 100% agreed that we shouldn't be exporting the implementation classes. But we should ensure it remains possible to pattern-match on kinds of DDSs, and that there is an ergonomic way to do so.

This workaround effectively means that consumers lose the ability to provide custom tree visualizers (a degree of freedom they have for the other DDSs), unless they know how to implement this same workaround..

Reply:
@jikim-msft would it be easier to just inline `new SharedTreeFactory().type` in our visualizer array declaration, rather than adding this type?